### PR TITLE
chore: update model card for SigLIP

### PR DESCRIPTION
**Pull request by saswatmeher**
# What does this PR do?

Update the model card for SigLIP to handle #36979 

Fixes #36979 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/siglip.md [hunk 1]
```diff
@@ -14,184 +14,102 @@ rendered properly in your Markdown viewer.
...
 ```python
->>> from transformers import SiglipModel
-
->>> model = SiglipModel.from_pretrained(
-...     "google/siglip-so400m-patch14-384",
-...     attn_implementation="sdpa",
-...     torch_dtype=torch.float16,
-...     device_map=device,
-... )
+from transformers import SiglipModel
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip.md:103]: Indent code here and lets show `flash_attention_2` since we're already using `"sdpa"` above.

```py
# pip install -U flash-attn --no-build-isolation

<code here>
```

File: docs/source/en/model_doc/siglip.md [hunk 2]
```diff
@@ -14,184 +14,102 @@ rendered properly in your Markdown viewer.
...
-
-
-## Expected speedups
-
-Below is an expected speedup diagram that compares inference time between the native implementation in transformers using `google/siglip-so400m-patch14-384` checkpoint in `float16` precision and the Flash Attention 2 / SDPA version of the model using different batch sizes.
-
-<div style="text-align: center">
-<img src="https://i.imgur.com/cWm4rsn.png">
-</div>
+- For the best speedups, we recommend loading the model in half-precision (e.g. `torch.float16` or `torch.bfloat16`).
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip.md:112]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Great job, thanks so much!

File: docs/source/en/model_doc/siglip.md [hunk 3]
```diff
@@ -14,184 +14,102 @@ rendered properly in your Markdown viewer.
...
-The original code can be found [here](https://github.com/google-research/big_vision/tree/main).
+The example below demonstrates how to generate similarity scores between texts and image(s) with [`Pipeline`] or the [`AutoModel`] class.
 
-## Usage example
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-There are 2 main ways to use SigLIP: either using the pipeline API, which abstracts away all the complexity for you, or by using the `SiglipModel` class yourself.
+```py
+from transformers import pipeline
```
[the remark this answers is not in the document]
Comment (saswatmeher) [reply docs/source/en/model_doc/siglip.md:44]: done

File: docs/source/en/model_doc/siglip.md [hunk 4]
```diff
@@ -14,184 +14,102 @@ rendered properly in your Markdown viewer.
...
-## Resources
-
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with SigLIP.
-
-- [Zero-shot image classification task guide](../tasks/zero_shot_image_classification)
-- Demo notebooks for SigLIP can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/SigLIP). 🌎
-
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+</hfoption>
+<hfoption id="AutoModel">
```
[the remark this answers is not in the document]
Comment (saswatmeher) [reply docs/source/en/model_doc/siglip.md:54]: done

File: docs/source/en/model_doc/siglip.md [hunk 5]
```diff
@@ -14,184 +14,102 @@ rendered properly in your Markdown viewer.
...
->>> probs = torch.sigmoid(logits_per_image) # these are the probabilities
->>> print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
+logits_per_image = outputs.logits_per_image
+probs = torch.sigmoid(logits_per_image) # these are the probabilities
+print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
 19.8% that image 0 is '2 cats'
 ```
 
+</hfoption>
+</hfoptions>
```
[the remark this answers is not in the document]
Comment (saswatmeher) [reply docs/source/en/model_doc/siglip.md:80]: done

File: docs/source/en/model_doc/siglip.md (hunk 1 above)
Comment (saswatmeher) [reply docs/source/en/model_doc/siglip.md]: done

**Review by stevhliu — APPROVED** [review]
Nice, thanks again! 🤗

File: docs/source/en/model_doc/siglip.md [hunk 6]
```diff
@@ -14,184 +14,116 @@ rendered properly in your Markdown viewer.
...
+probs = torch.sigmoid(logits_per_image)
+print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
 ```
 
-## Resources
-
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with SigLIP.
-
-- [Zero-shot image classification task guide](../tasks/zero_shot_image_classification)
-- Demo notebooks for SigLIP can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/SigLIP). 🌎
```
Comment (qubvel) [inline docs/source/en/model_doc/siglip.md:112]: @stevhliu, I'm curious, should we remove links to additional resources?

Comment (stevhliu) [reply docs/source/en/model_doc/siglip.md:112]: Hey, great question! I think we should remove these additional resources because imo the docs should only contain official and maintained content. Everything you need to know should ideally be in the docs, and additional resources linking to extra/nice-to-know/supplemental content should be kept somewhere else. When a notebook or blog from the community gets outdated or stops working as expected, its a bit harder to maintain and can also add noise and clutter.

These additional resources are still useful and there is definitely a place for them though! I'm experimenting a bit with creating a "[Transformers model resources](https://huggingface.co/spaces/stevhliu/model-resources)" Space that will hold all these additional resources. Community members can open PRs to add a notebook or open fixes for it there.

Let me know if that helps answer your question! :)

Comment (qubvel) [reply docs/source/en/model_doc/siglip.md:112]: @stevhliu thanks for answering, having a space indeed a cool idea! However, in my opinion, from a user perspective, it would be super nice to have some "further steps" when you land on the documentation page - so you don't need to search for additional resources and can continue exploring the model. In transformers, we are trying to keep the external API stable, so most of the resources are still relevant even though they are not maintained. Considering the case above

> \[Zero-shot image classification task guide](../tasks/zero_shot_image_classification)

This one is actually pointing to our internal guide

> Demo notebooks for SigLIP can be found \[here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/SigLIP). 🌎

This one is also still working

So, I would be more careful about removing useful materials and doing it only in case some users complain.

cc @NielsRogge as well, because you were the one who added a lot of resources
